src/test/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControlTest.java

```diff
@@ -224,6 +257,200 @@ void shouldGenerateAccessProfilesByCaseDetails() {
         verify(accessProfileService).generateAccessProfiles(anyList(), anyList());
     }
 
+    @Test
+    void shouldGenerateAccessProfilesByCaseAccessCategory_MatchCommaSeparatedLine() throws JsonProcessingException {
+        doReturn(USER_ID).when(securityUtils).getUserId();
+        CaseDetails caseDetails = new CaseDetails();
+        caseDetails.setCaseTypeId(CASE_TYPE_1);
+        caseDetails.setData(generateCaseData("UNSPEC_CLAIM"));
+
+        RoleAssignments roleAssignments = new RoleAssignments();
+        doReturn(roleAssignments).when(roleAssignmentService).getRoleAssignments(anyString());
+
+        doReturn(filteredRoleAssignments).when(roleAssignmentsFilteringService)
+            .filter(any(RoleAssignments.class), any(CaseDetails.class));
+
+        CaseTypeDefinition caseTypeDefinition = createCaseTypeDefinition(ROLE_NAME_1);
+        doReturn(caseTypeDefinition).when(caseDefinitionRepository).getCaseType(CASE_TYPE_1);
+
+        doReturn(Collections.emptyList()).when(filteredRoleAssignments).getFilteredMatchingRoleAssignments();
+
+        doReturn(false).when(applicationParams).getEnablePseudoRoleAssignmentsGeneration();
+
+        doReturn(dummyAccessProfiles).when(accessProfileService).generateAccessProfiles(anyList(), anyList());
+
+        Set<AccessProfile> accessProfiles = defaultCaseDataAccessControl
+            .generateAccessProfilesByCaseDetails(caseDetails);
+
+        assertNotNull(accessProfiles);
+        assertEquals(3, accessProfiles.size());
+        Assertions.assertTrue(accessProfiles.stream().filter(a -> a.getCaseAccessCategories() != null)
+            .allMatch(a -> a.getCaseAccessCategories().contains("UNSPEC_CLAIM")));
```

Comment:
Optional (and applies to all new tests): Might be better to assert the access profiles that pass through based on their `accessProfile`, since that's unique in the data setup. From that test I assume it's `ap_1`, `ap_2` and `ap_4` that are in the result - while you can infer that from this assertion by comparing with the data, being more explicit could be useful.

Reply:
done